components/brave_wallet_ui/page/screens/onboarding/restore-from-recovery-phrase/restore-from-recovery-phrase.tsx

```diff
@@ -160,100 +197,152 @@ export const OnboardingRestoreFromRecoveryPhrase = () => {
 
   // render
   if (isCreatingWallet) {
-    return <CreatingWallet />
+    return <OnboardingCreatingWallet />
   }
 
   return (
-    <CenteredPageLayout>
-      <MainWrapper>
-        <StyledWrapper>
-          <OnboardingStepsNavigation preventSkipAhead />
-
-          <TitleAndDescriptionContainer>
-            <Title>
-              {getLocale(
-                currentStep === 'password'
-                  ? 'braveWalletCreatePasswordTitle'
-                  : 'braveWalletRestoreMyBraveWallet'
-              )}
-            </Title>
-            <Description>
-              {getLocale(
-                currentStep === 'password'
-                  ? 'braveWalletCreatePasswordDescription'
-                  : 'braveWalletRestoreMyBraveWalletInstructions'
-              )}
-            </Description>
-          </TitleAndDescriptionContainer>
-
-          {currentStep === 'phrase' && (
-            <>
-              <PhraseCard>
-                <RecoveryInput
-                  onChange={onRecoveryInputChanged}
-                  onKeyDown={handleKeyDown}
-                  onToggleShowPhrase={setIsPhraseShown}
+    <OnboardingContentLayout
+      title={title}
+      subTitle={subTitle}
+    >
+      {currentStep === 'phrase' && (
+        <>
+          <Column
+            alignItems='flex-end'
+            justifyContent='center'
+            margin='52px 0 28px'
+          >
+            <RecoveryPhraseContainer phraseLength={recoveryPhraseLength}>
+              {Array.from({ length: recoveryPhraseLength }, (_, index) => (
+                <Input
+                  key={index}
+                  mode='filled'
+                  placeholder={getLocale(
+                    'braveWalletRecoveryPhraseWord'
+                  ).replace('$1', `#${index + 1}`)}
+                  value={phraseWords[index] || ''}
+                  size='small'
+                  onChange={(e) => onPhraseWordChange(index, e.value)}
+                  onInput={(e) => onPhraseWordChange(index, e.value)}
```

Comment:
can we clear the user's clipboard when they paste like we do in the current onboarding?

Reply:
@josheleonard  Done